src/TensorNetwork.jl

```diff
@@ -261,20 +281,52 @@ function __check_index_sizes(tn)
     return true
 end
 
-const is_unsafe_region = ScopedValue(false) # global ScopedValue for the unsafe region
+Base.in(tn::TensorNetwork, uc::UnsafeContext) = tn ∈ values(uc)
 
-macro unsafe_region(tn, block)
+# Function to get the current UnsafeContext
+function current_unsafe_context()
+    if isempty(Tenet._unsafe_context_stack[])
+        return nothing
+    else
+        return Tenet._unsafe_context_stack[][end]
+    end
+end
+
+# Define the @unsafe_region macro
```

Comment:
this comment is a lil bit auto-descriptive? maybe we can just remove it or add a docstring

Reply:
Maybe its better to remove it. I prefer not to have a `docstring` here since this is an internal operation, for now.